Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -134,4 +306,77 @@ extension PreplannedMapArea: PreplannedMapAreaProtocol {
     }
+    
+    var id: PortalItem.ID? {
+        portalItem.id
+    }
+}
+
+private extension FileManager {
+    static let mmpkPathExtension: String = "mmpk"
+    static let offlineMapAreasPath: String = "OfflineMapAreas"
+    static let packageDirectoryPath: String = "Package"
+    static let preplannedDirectoryPath: String = "Preplanned"
+    
+    /// The path to the documents folder.
+    var documentsDirectory: URL {
+        URL.documentsDirectory
+    }
+    
+    /// The path to the offline map areas directory within the documents directory.
+    var offlineMapAreasDirectory: URL {
+        documentsDirectory.appending(
+            path: Self.offlineMapAreasPath,
+            directoryHint: .isDirectory
+        )
+    }
+    
+    /// The path to the web map directory for a specific portal item.
+    /// - Parameter portalItemID: The ID of the web map portal item.
+    func webMapDirectory(forPortalItemID portalItemID: String) -> URL {
+        offlineMapAreasDirectory.appending(path: portalItemID, directoryHint: .isDirectory)
+    }
+    
+    /// The path to the preplanned map areas directory for a specific portal item.
+    /// - Parameter portalItemID: The ID of the web map portal item.
+    func preplannedDirectory(forPortalItemID portalItemID: String) -> URL {
+        webMapDirectory(forPortalItemID: portalItemID).appending(
+            path: Self.preplannedDirectoryPath,
+            directoryHint: .isDirectory
+        )
+    }
+    
+    /// The path to the download directory for the preplanned map area metadata and mobile map package.
+    /// - Parameters:
+    ///   - portalItemID: The ID of the web map portal item.
+    ///   - preplannedMapAreaID: The ID of the preplanned map area.
+    func downloadDirectory(forPortalItemID portalItemID: String, preplannedMapAreaID: String) -> URL {
+        preplannedDirectory(forPortalItemID: portalItemID)
+            .appending(
+                path: preplannedMapAreaID,
+                directoryHint: .isDirectory
+            )
+    }
+    
+    /// The path to the package directory for the preplanned map area mobile map package.
+    /// - Parameters:
+    ///   - portalItemID: The ID of the web map portal item.
+    ///   - preplannedMapAreaID: The ID of the preplanned map area.
+    func packageDirectory(forPortalItemID portalItemID: String, preplannedMapAreaID: String) -> URL {
+        downloadDirectory(forPortalItemID: portalItemID, preplannedMapAreaID: preplannedMapAreaID)
+            .appending(
+                component: Self.packageDirectoryPath,
+                directoryHint: .isDirectory
+            )
+    }
+    
+    /// The path to the mobile map package file for the preplanned map area mobile map package.
+    /// - Parameters:
+    ///   - portalItemID: The ID of the web map portal item.
+    ///   - preplannedMapAreaID: The ID of the preplanned map area.
+    func mmpkDirectory(forPortalItemID portalItemID: String, preplannedMapAreaID: String) -> URL {
```

Comment:
this directory structure seems un-necessary:

```
Documents/OfflineMapAreas/<portal item id>/Preplanned/<preplanned area id>/Package/<preplanned area id>.mmpk
```

I think this would suffice:

```
Documents/OfflineMapAreas/<portal item id>/Preplanned/<preplanned area id>/<preplanned area id>.mmpk
```